File: ecmascript/pig-latin/babel-output/pig-latin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var vowelRegex = RegExp(/(^|\s+)(a|e|i|o|u|yt|xr)(\w+)/g);
var consonantRegex = RegExp(/(^|\s+)(ch|qu|thr|th|sch|yt|\wqu|\w)(\w+)/g);

var vowelReplacement = "$1$2$3ay";
var consonantReplacement = "$1$3$2ay";

var PigLatin = (function () {
    function PigLatin() {
        _classCallCheck(this, PigLatin);
    }

    _createClass(PigLatin, [{
        key: "translate",
        value: function translate(sentence) {
            if (vowelRegex.test(sentence)) return sentence.replace(vowelRegex, vowelReplacement);else return sentence.replace(consonantRegex, consonantReplacement);
        }
    }]);

    return PigLatin;
})();

exports["default"] = PigLatin;
module.exports = exports["default"];

File: v1/java/pig-latin/src/main/java/PigLatin.java
public class PigLatin {

    private static final String VOWEL_REGEX = "(?<begin>^|\\s+)(?<vowel>a|e|i|o|u|yt|xr)(?<rest>\\w+)";
    private static final String CONSONANT_REGEX = "(?<begin>^|\\s+)(?<consonant>ch|qu|thr|th|sch|yt|\\wqu|\\w)(?<rest>\\w+)";

    private static final String VOWEL_REPLACEMENT = "${begin}${vowel}${rest}ay";
    private static final String CONSONANT_REPLACEMENT = "${begin}${rest}${consonant}ay";

    public static String translate(String input) {
        if (input.matches(VOWEL_REGEX)) {
            return input.replaceAll(VOWEL_REGEX, VOWEL_REPLACEMENT);
        }

        return input.replaceAll(CONSONANT_REGEX, CONSONANT_REPLACEMENT);
    }
}
